runtime/worker.rs

```diff
@@ -553,12 +555,31 @@ impl MainWorker {
 
       event_loop_result = self.run_event_loop(false) => {
         event_loop_result?;
-
         receiver.await
       }
     }
   }
 
+  /// Run the event loop up to a given duration.
```

Comment:
I might promote this to deno_core but I would like to let it bake in deno for a bit